src/resources/parser/glb-parser.js

```diff
@@ -1201,99 +1285,24 @@ const createMaterial = function (gltfMaterial, textures, flipV) {
         material.cull = CULLFACE_BACK;
     }
 
-    if (gltfMaterial.hasOwnProperty('extensions') &&
-        gltfMaterial.extensions.hasOwnProperty('KHR_materials_clearcoat')) {
-        const ccData = gltfMaterial.extensions.KHR_materials_clearcoat;
-
-        if (ccData.hasOwnProperty('clearcoatFactor')) {
-            material.clearCoat = ccData.clearcoatFactor * 0.25; // TODO: remove temporary workaround for replicating glTF clear-coat visuals
-        } else {
-            material.clearCoat = 0;
-        }
-        if (ccData.hasOwnProperty('clearcoatTexture')) {
-            const clearcoatTexture = ccData.clearcoatTexture;
-            material.clearCoatMap = textures[clearcoatTexture.index];
-            material.clearCoatMapChannel = 'r';
-
-            extractTextureTransform(clearcoatTexture, material, ['clearCoat']);
-        }
-        if (ccData.hasOwnProperty('clearcoatRoughnessFactor')) {
-            material.clearCoatGlossiness = ccData.clearcoatRoughnessFactor;
-        } else {
-            material.clearCoatGlossiness = 0;
-        }
-        if (ccData.hasOwnProperty('clearcoatRoughnessTexture')) {
-            const clearcoatRoughnessTexture = ccData.clearcoatRoughnessTexture;
-            material.clearCoatGlossMap = textures[clearcoatRoughnessTexture.index];
-            material.clearCoatGlossMapChannel = 'g';
-
-            extractTextureTransform(clearcoatRoughnessTexture, material, ['clearCoatGloss']);
-        }
-        if (ccData.hasOwnProperty('clearcoatNormalTexture')) {
-            const clearcoatNormalTexture = ccData.clearcoatNormalTexture;
-            material.clearCoatNormalMap = textures[clearcoatNormalTexture.index];
-
-            extractTextureTransform(clearcoatNormalTexture, material, ['clearCoatNormal']);
+    // Provide list of supported extensions and their functions
+    const extensions = {
+        "KHR_materials_pbrSpecularGlossiness": pbrSpecGlossiness,
+        "KHR_materials_clearcoat": clearCoat,
+        "KHR_materials_unlit": unlit,
+        "KHR_materials_specular": metallicF0
```

Comment:
perhaps give the function names some common naming .. for example instead of `clearCoat` have `extensionClearCoat`.

Reply:
Good point, done.